src/servicesHost.ts

```diff
@@ -496,12 +520,22 @@ function resolveModuleName(
     // tslint:disable-next-line:no-empty
   } catch (e) {}
 
-  const tsResolution = compiler.resolveModuleName(
-    moduleName,
-    containingFile,
-    compilerOptions,
-    moduleResolutionHost
-  );
+  const tsResolution =
+    customResolveModuleName !== undefined
+      ? customResolveModuleName(
+          moduleName,
+          containingFile,
+          compilerOptions,
+          moduleResolutionHost,
+          applyTsResolver.bind(null, compiler)
```

Comment:
Seeing a `.bind` my perf spider sense is tingling!  

https://stackoverflow.com/questions/42117911/lambda-functions-vs-bind-memory-and-performance

I wonder if we could switch this to a lamba instead?  It'll be slightly more verbose but hopefully should perform better...